Disc/Networking/APIClient.swift

```diff
@@ -12,4 +12,10 @@ public struct APIClient {
     public init(token: String) {
         self.token = token
     }
+    
+    /// Create an APIClient with no access token.
+    ///
+    public init() {
+        self.token = ""
+    }
```

Comment:
I think making an access token required for all of these operations has a nice simplicity to it. We can provide ways to make it easy to get a token, like `getAccessToken`.

This pull request is already evidence of ways in which obtaining an access token can change, but that isn't a concern for this interface so it shouldn't require any changes in order to keep up.